crates/re_types/src/archetypes/mesh3d_ext.rs

```diff
@@ -0,0 +1,77 @@
+use super::Mesh3D;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Mesh3DError {
+    #[error("Indices array length must be divisible by 3 (triangle list), got {0}")]
+    IndicesNotDivisibleBy3(usize),
+
+    #[error("No indices were specified, so the number of positions must be divisible by 9 [(xyz xyz xyz), …], got {0}")]
+    PositionsAreNotTriangles(usize),
+
+    #[error("Index out of bounds: got index={index} with {num_vertices} vertices")]
+    IndexOutOfBounds { index: u32, num_vertices: usize },
+
+    #[error(
+        "Positions & normals array must have the same length, \
+        got positions={0} vs. normals={1}"
+    )]
+    MismatchedPositionsNormals(usize, usize),
+}
+
+impl Mesh3D {
+    pub fn sanity_check(&self) -> Result<(), Mesh3DError> {
+        let num_vertices = self.num_vertices();
+
+        if let Some(triangle_indices) = self
+            .mesh_properties
+            .as_ref()
+            .and_then(|props| props.triangle_indices.as_ref())
+        {
+            if triangle_indices.len() % 3 != 0 {
+                return Err(Mesh3DError::IndicesNotDivisibleBy3(triangle_indices.len()));
+            }
+
+            for &index in triangle_indices.iter() {
+                if num_vertices <= index as usize {
+                    return Err(Mesh3DError::IndexOutOfBounds {
+                        index,
+                        num_vertices,
+                    });
+                }
+            }
+        } else if self.vertex_positions.len() % 9 != 0 {
+            return Err(Mesh3DError::PositionsAreNotTriangles(
+                self.vertex_positions.len(),
+            ));
+        }
+
+        if let Some(normals) = &self.vertex_normals {
+            if normals.len() != self.vertex_positions.len() {
+                return Err(Mesh3DError::MismatchedPositionsNormals(
+                    self.vertex_positions.len(),
+                    normals.len(),
+                ));
+            }
+        }
+
+        Ok(())
+    }
+
+    #[inline]
+    pub fn num_vertices(&self) -> usize {
+        self.vertex_positions.len()
+    }
+
+    #[inline]
+    pub fn num_triangles(&self) -> usize {
+        if let Some(triangle_indices) = self
+            .mesh_properties
+            .as_ref()
+            .and_then(|props| props.triangle_indices.as_ref())
+        {
+            triangle_indices.len()
```

Comment:
Note on the naming mentioned previously: I'm unsure here if this is a bug or not. It depends on if `triangle_indices` is `[u32]` or `[UVec3]` here, which I can't tell from reading the code.

I suggest this naming, unless someone comes up with something better:

```
/// One element per traingle.
triangle_indices: [UVec3],

/// Three elements for each triangle.
vertex_indices: [u32],
```